spec/services/mediator_spec.rb

```diff
@@ -0,0 +1,47 @@
+RSpec.describe Mediator do
+  subject { described_class }
+
+  let(:collection) { create(:collection) }
+  let(:user) { create(:user) }
+  let(:file) { nil }
+  let(:filename) { nil }
+  let(:link) { nil }
+  let(:chunking_method) { :basic }
+  let(:chunking_profile) { create(:chunking_profile, method: chunking_method, size: 800) }
+  let(:doc) { Document.new(state: :created, link:, file:, filename:, chunking_profile:, collection:, user:) }
+
+  describe "#ingest document" do
+    context "with web link" do
+      let(:link) { "https://en.wikipedia.org/wiki/Tabloid_journalism" }
+
+      it "fetches web page" do
+        expect do
+          subject.ingest(doc)
+        end.to change(FetchWebDocumentJob.jobs, :size).by(1)
+        Sidekiq::Worker.clear_all
+      end
+
+      context "when web page has been fetched" do
+        it "chunks document" do
+          doc.save
+          subject.ingest(doc)
+
+          expect { FetchWebDocumentJob.drain }.to change(ChunkDocumentJob.jobs, :size).by(1)
+          Sidekiq::Worker.clear_all
+        end
+      end
+    end
+
+    context "with filename" do
+      let(:file) { fixture_file_upload("gnu_manifesto.md", 'application/html') }
+      let(:filename) { "spec/fixtures/files/gnu_manifesto.md" }
+
+      it "chunks document" do
+        expect do
+          subject.ingest(doc)
+        end.to change(ChunkDocumentJob.jobs, :size).by(1)
+        Sidekiq::Worker.clear_all
```

Comment:
To me, duplicating it at the end of every example is worse than putting in an `after` block, because now we have the opportunity to forget it on every example, instead of just when creating a new spec!

The `after` block is perfectly dependable - I didn't mean to cast aspersions on it. What I was commenting on is that if we test job enqueueing this way, and we add a new spec, now we need to remember the `after` block or we introduce flakiness. Whereas testing jobs the way I suggested there is no risk of flakiness and no need to remember to clear a job queue in each spec.

Reply:
Ah, I misunderstood. Still getting used to the conventions.

Re sidekiq jobs, from what I can tell, the rspect support by default queues up jobs in the job class. It does mean that un-drained job will accumulate, and possibly across examples. I ran into issues with the job counts. So makes sense if I can avoid relying on those in the first place.